src/Semantic/Api/Diffs.hs

```diff
@@ -82,84 +79,78 @@ diffGraph blobs = do
 
-instance (ConstructorName syntax, Diffable syntax, Eq1 syntax, Hashable1 syntax, Traversable syntax) => SExprDiff (Term syntax) where
-  sexprDiff = serialize (SExpression ByConstructorName)
 
+class SExprDiff term where
+  sexprDiff :: (Carrier sig m, Member (Reader Config) sig, Member Telemetry sig, MonadIO m) => Edit (Blob, term Loc) (Blob, term Loc) -> m Builder
 
-showDiffParsers :: Map Language (SomeParser ShowDiff Loc)
-showDiffParsers = aLaCarteParsers
+instance (DiffTerms term, ConstructorName (Syntax term), Foldable (Syntax term), Functor (Syntax term)) => SExprDiff term where
+  sexprDiff = serialize (SExpression ByConstructorName) <=< diffTerms
 
-class DiffTerms term => ShowDiff term where
-  showDiff :: (Carrier sig m, Member (Reader Config) sig) => DiffFor term Loc Loc -> m Builder
 
-instance (Diffable syntax, Eq1 syntax, Hashable1 syntax, Show1 syntax, Traversable syntax) => ShowDiff (Term syntax) where
-  showDiff = serialize Show
+class ShowDiff term where
+  showDiff :: (Carrier sig m, Member (Reader Config) sig, Member Telemetry sig, MonadIO m) => Edit (Blob, term Loc) (Blob, term Loc) -> m Builder
 
+instance (DiffTerms term, Foldable (Syntax term), Show1 (Syntax term)) => ShowDiff term where
+  showDiff = serialize Show <=< diffTerms
 
--- | Parse a 'BlobPair' using one of the provided parsers, diff the resulting terms, and run an action on the abstracted diff.
---
--- This allows us to define features using an abstract interface, and use them with diffs for any parser whose terms support that interface.
-diffWith
-  :: (forall term . c term => DiffTerms term, DiffEffects sig m)
-  => Map Language (SomeParser c Loc)                            -- ^ The set of parsers to select from.
-  -> (forall term . c term => DiffFor term Loc Loc -> m output) -- ^ A function to run on the computed diff. Note that the diff is abstract (it’s the diff type corresponding to an abstract term type), but the term type is constrained by @c@, allowing you to do anything @c@ allows, and requiring that all the input parsers produce terms supporting @c@.
-  -> BlobPair                                                   -- ^ The blob pair to parse.
-  -> m output
-diffWith parsers render = parsePairWith parsers (render <=< diffTerms)
 
-diffTerms :: (DiffTerms term, Member Telemetry sig, Carrier sig m, MonadIO m)
-  => Edit (Blob, term ann) (Blob, term ann) -> m (DiffFor term ann ann)
+diffTerms :: (DiffTerms term, Foldable (Syntax term), Member Telemetry sig, Carrier sig m, MonadIO m)
+  => Edit (Blob, term ann) (Blob, term ann) -> m (Diff (Syntax term) ann ann)
 diffTerms terms = time "diff" languageTag $ do
   let diff = diffTermPair (bimap snd snd terms)
   diff <$ writeStat (Stat.count "diff.nodes" (bilength diff) languageTag)
   where languageTag = languageTagForBlobPair blobs
         blobs = bimap fst fst terms
+
+diffParsers :: Map Language (SomeParser Anything Loc)
+diffParsers = aLaCarteParsers
```

Comment:
We no longer have to define separate diff parsers for each language; it’s a reasonable optimization to just lump everything together (for diffing—the `.Terms` module can’t really do this because some of the operations are defined for precise terms as well, and some are not).